src/main/java/com/pmm/budgetcalculator/expensing/service/ExpenseService.java

```diff
@@ -26,9 +26,10 @@ public Expense getExpenseById(Long id) {
     public Expense createExpense(Expense expense) {
         return expenseRepository.save(expense);
     }
-        public void removeExpenseById(Long id){
-            if (expenseRepository.existsById(id)) {
-                expenseRepository.deleteById(id);
-            }
+
+    public void removeExpenseById(Long id) {
+        if (expenseRepository.existsById(id)) {
```

Comment:
What if it does not exists? In your controller method returns HTTP code OK. So now no matter expense was found and deleted or not I as a user will see HTTP OK and think everything is good. Take a look at getExpenseById - you should throw exception in else block.